crates/types/src/config/mod.rs

```diff
@@ -88,8 +92,17 @@ pub fn node_filepath(filename: &str) -> PathBuf {
 pub fn reset_base_temp_dir() -> PathBuf {
     let mut guard = NODE_BASE_DIR.write().unwrap();
     let new = tempfile::TempDir::new().unwrap();
-    *guard = new;
-    PathBuf::from(guard.path())
+    let path = PathBuf::from(new.path());
+    *guard = TempOrPath::Temp(new);
+    path
+}
+
+#[cfg(any(test, feature = "test-util"))]
+pub fn reset_base_temp_dir_and_retain() -> PathBuf {
```

Comment:
Maybe add comment explaining that retain means retaining the directory beyond the lifetime of the process.